array.c

```diff
@@ -6998,22 +7013,59 @@ flatten(VALUE ary, int level)
 
 /*
  *  call-seq:
- *     ary.flatten!        -> ary or nil
- *     ary.flatten!(level) -> ary or nil
+ *    array.flatten! -> self or nil
+ *    array.flatten!(level) -> self or nil
+ *
+ *  Replaces each nested \Array in +self+ with the elements from that \Array;
+ *  returns +self+ if any changes, +nil+ otherwise.
  *
- *  Flattens +self+ in place.
+ *  Argument +level+, if given, must be an
+ *  {Integer-convertible object}[doc/implicit_conversion_rdoc.html#label-Integer-Convertible+Objects].
+ *
+ *  With no argument, flattens all levels:
```

Comment:
Let's avoid duplication and state `With no argument or with a negative argument, flattens all levels`, since the behavior is the same.  Then we can avoid the separate section for a negative argument below.  Same for flatten below.